### PR TITLE
Support time travel for getMetadata in OSS server

### PR DESCRIPTION
**Pull request by linzhou-db**
Support time travel for getMetadata in OSS server, tested with new unit tests

### CODE REVIEW

File: server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala [hunk 1]
```diff
@@ -288,20 +288,30 @@ class DeltaSharingService(serverConfig: ServerConfig) {
...
       @Param("share") share: String,
       @Param("schema") schema: String,
-      @Param("table") table: String): HttpResponse = processRequest {
+      @Param("table") table: String,
+      @Param("version") @Nullable version: java.lang.Long,
+      @Param("timestamp") @Nullable timestamp: String): HttpResponse = processRequest {
     import scala.collection.JavaConverters._
+    if (version != null && version < 0) {
+      throw new DeltaSharingIllegalArgumentException("version cannot be negative.")
+    }
```
Comment (charlenelyu-db) [inline server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala:297]: shall we add multiple params check?

File: server/src/test/scala/io/delta/sharing/server/DeltaSharingServiceSuite.scala [hunk 2]
```diff
@@ -1930,6 +1941,98 @@ class DeltaSharingServiceSuite extends FunSuite with BeforeAndAfterAll {
...
+    )
+  }
+
+  def verifyMetadata(version: Long, expectedMetadata: Metadata): Unit = {
+    val response = getMetadataResponse(version)
+    val Array(protocol, metadata) = response.split("\n")
+    val expectedProtocol = Protocol(minReaderVersion = 1)
+    assert(expectedProtocol.wrap == JsonUtils.fromJson[SingleAction](protocol))
+    assert(expectedMetadata.wrap == JsonUtils.fromJson[SingleAction](metadata))
+  }
```
Comment (charlenelyu-db) [inline server/src/test/scala/io/delta/sharing/server/DeltaSharingServiceSuite.scala:1970]: Can we take `val response = getMetadataResponse(..)` out of `verifyMetadata()` and make `verifyMetadata()` only responsible for comparing response string with the expectation? 
Like `verifyMetadata(getMetadataResponse(0), StreamingNotnullToNull.metadataV0)`. 

By doing so we don't need to keep two versions of `verifyMetadata()`, and it's more readable.

**Review by charlenelyu-db — APPROVED** [review]

File: server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala (hunk 1 above)
Comment (linzhou-db) [reply server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala:297]: good catch, added.

File: server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala [hunk 3]
```diff
@@ -288,20 +288,30 @@ class DeltaSharingService(serverConfig: ServerConfig) {
...
     val capabilitiesMap = getDeltaSharingCapabilitiesMap(
       req.headers().get(DELTA_SHARING_CAPABILITIES_HEADER)
     )
     val tableConfig = sharedTableManager.getTable(share, schema, table)
+    if ((version != null || timestamp != null ) && !tableConfig.historyShared) {
```
[the remark this answers is not in the document]
Comment (linzhou-db) [reply server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala:302]: nice! no bin/scalafmt for oss developement, and this is not catched by scalastyle...

File: server/src/test/scala/io/delta/sharing/server/DeltaSharingServiceSuite.scala (hunk 2 above)
Comment (linzhou-db) [reply server/src/test/scala/io/delta/sharing/server/DeltaSharingServiceSuite.scala:1970]: updated.

**Pull request merged**